src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -950,187 +953,205 @@ public override ValueTask RemoveKeys(IEnumerable<TrackedObjectKey> keys)
+                        }
+                        catch (OperationCanceledException)
+                        {
+                            this.partition.EventDetailTracer?.TraceEventProcessingDetail($"query {queryId} attempt {attempt:o} cancelled");
+                            goto done;
+                        }
+                        catch (Exception exception) when (this.terminationToken.IsCancellationRequested && !Utils.IsFatal(exception))
+                        {
+                            this.partition.EventDetailTracer?.TraceEventProcessingDetail($"query {queryId} attempt {attempt:o} cancelled due to partition termination");
+                            cancellationTokenSource.Cancel();
+                            goto done;
+                        }
+                        catch (Exception e)
                         {
-                            if (matched >= pageLimit.Value)
+                            this.partition.EventTraceHelper.TraceEventProcessingWarning($"query {queryId} attempt {attempt:o} enumeration failed with exception {e}");
+                            cancellationTokenSource.Cancel();
+                            goto done;
+                        }
+
+                        if (orchestrationState != null && instanceQuery.Matches(orchestrationState))
+                        {
+                            matched++;
+                            this.partition.EventDetailTracer?.TraceEventProcessingDetail($"match instance {queryRequests.Current}");
+                            yield return (position, orchestrationState);
+
+                            if (pageLimit.HasValue)
                             {
-                                cancellationTokenSource.Cancel();
-                                break;
+                                if (matched >= pageLimit.Value)
+                                {
+                                    cancellationTokenSource.Cancel();
+                                    break;
+                                }
+                            }
+                            else
+                            {
+                                pageCapacity.Release();
                             }
                         }
                         else
                         {
-                            leftToFill.Release();
+                            pageCapacity.Release();
                         }
                     }
-                    else
-                    {
-                        leftToFill.Release();
-                    }
                 }
+            pageDone:
+                yield return (position, null);
+                ReportProgress("completed-page");
+            done:
+                yield break;
+            }
+            finally
+            {
+                cancellationTokenSource.Cancel();
+                await produceFasterReadsTask;
```

Comment:
For consistency this await should use `ConfigureAwait(false)` also, I think.

And perhaps we could add a `ReportProgress` after this await? So we can detect cases where the await hangs or where the finally block does not get called (finally block in method with `yield` gets called only if caller correctly disposes enumerator).

Reply:
Added the ConfigureAwait(false)

Reply:
And added the ReportProgress on my last commit